rules/no-native-promise-helpers.js

```diff
@@ -26,6 +28,29 @@ module.exports = {
         }
       },
 
+      MethodDefinition(node) {
+        if (parentTask) return;
+        let { value, decorators } = node;
+        if (!decorators) return;
+        if (!value) return;
+        if (!value.generator) return;
+
+        for (let decorator of node.decorators) {
+          if (
+            decorator.expression.type === 'Identifier' &&
+            TASK_TYPES.includes(decorator.expression.name)
+          ) {
+            parentTask = node;
+          }
+        }
```

Comment:
or maybe even better, extract a shared `hasTaskDecorator(node)` function:

```js
function hasTaskDecorator(node) {
  if (!node.decorators) return false;

  for (let decorator of node.decorators) {
    let { expression } = decorator;
    if (
      expression &&
      expression.type === 'Identifier' &&
      TASK_TYPES.includes(expression.name)
    ) {
      return true;
    }
  }

  return false;
}
```

or using `some()`:

```js
function hasTaskDecorator(node) {
  if (!node.decorators) return false;

  return node.decorators.some(decorator => {
    let { expression } = decorator;
    return expression &&
      expression.type === 'Identifier' &&
      TASK_TYPES.includes(expression.name);
  });
}
```

Reply:
Yeah that is a good point, will add it in my next 20%

Reply:
@Turbo87 I have updated it now to use a helper, let me know what you think